_LowLevelCode/CPP/cpp_communicator/cpp_communicator.cpp

```diff
@@ -93,7 +93,7 @@ Outputs: -- empty matrix.
 
 void mexFunction(int nlhs, mxArray* plhs[], int nrhs, const mxArray* prhs[])
 {
-    if (nrhs == 0 && nlhs == 1) {
+    if (nrhs == 0 && (nlhs == 0 || nlhs == 1)) {
```

Comment:
if `nlhs==0` does `plhs[0]` exist, should we just return in that case?

Reply:
If `nlhs == 0` and we define `plhs[0]` then the version will be assigned to `ans`. Or, we could just return and nothing is assigned to `ans`, it's up to us

Reply:
Ahh I see -- that seems like a reasonable behaviour then. I don't really have any strong feelings, just worried you were assigning something to a potentially empty array pointer.

In any case the same change needs to go into `get_ascii_file.cpp`